clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -161,7 +162,9 @@
         new NotControllerException("This is not the correct controller for this cluster.")),
     INVALID_REQUEST(42,
         new InvalidRequestException("This most likely occurs because of a request being malformed by the client library or" +
-            " the message was sent to an incompatible broker. See the broker logs for more details."));
+            " the message was sent to an incompatible broker. See the broker logs for more details.")),
+    MESSAGE_FORMAT_TOO_OLD(43,
```

Comment:
Would something like "UnsupportedOperation" or "UnsupportedForMessageFormat" be nearer the mark?